server/app/api/access_control/access_control_service.py

```diff
@@ -110,7 +110,7 @@ def has_write_access(self, user_id: int | None, repo_id: int) -> bool:
         # Case 7: Repo is in org and org has team permissions for that repo.
 
         for team_permission in team_permissions:
-            if team_permission.permission in [TeamPermissionKind.read_write, TeamPermissionKind.admin]:
+            if team_permission.kind in [TeamPermissionKind.read_write, TeamPermissionKind.admin]:
                 if self.team_repo.find_member(team_permission.team_id, user_id) is not None:
                     return True
```

Comment:
Good catch, I guess I missed this because add_dummy_data() doesn't create teams or team permissions, so this for-loop had never executed when I was doing docker push manual testing.